### PR TITLE
Add django-ansible-base

### PR DESCRIPTION
**Pull request by jessicamack**
##### SUMMARY

Add django-ansible-base to AWX

##### ISSUE TYPE
 
 - New or Enhanced Feature

##### COMPONENT NAME

 - API

##### AWX VERSION

```
awx: 23.5.2.dev1+g804a8b1c1e.d20231205
```

##### ADDITIONAL INFORMATION

Contains work from #14637 

```

```

### CODE REVIEW

**Comment by shanemcd** [comment]
Tests are failing with:

image

**Comment by AlanCoding** [comment]
I'm still seeing installer failures of

> AttributeError: module 'tarfile' has no attribute 'LinkOutsideDestinationError'

and I don't see this in runs for other branches, so I am starting to wonder if it's related to the changes here.

**Comment by AlanCoding** [comment]
^ no, never mind, that error is happening with lots of runs.

**Comment by gundalow** [comment]
Rebase your PR and `bulk_job_launch` CI jobs should pass.

**Review by shanemcd — APPROVED** [review]

**Review by gundalow — APPROVED** [review]

**Comment by AlanCoding** [comment]
Need to back up - from integration testing we have a test case where:
1. organization is created
2. new user is created
3. user is given auditor_role to the organization
4. As that user, GET /api/v2/organizations/N/

Unexpectedly, this gives a 403 error. This also seems to correlate with the upgrade of the `social-auth-app-django` to the 5.1.0 version specifically. Why that, I'm still not sure.

Mechanically, the problem seems to be because `request.user` is no longer a `User` object, but instead, a `SimpleLazyObject`. That breaks type checking.

I tested this patch as a fix:

```diff
diff --git a/awx/main/models/mixins.py b/awx/main/models/mixins.py
index 3ab32ba810..8f7ed9ade3 100644
--- a/awx/main/models/mixins.py
+++ b/awx/main/models/mixins.py
@@ -64,7 +64,7 @@ class ResourceMixin(models.Model):
 
     @staticmethod
     def _accessible_pk_qs(cls, accessor, role_field, content_types=None):
-        if type(accessor) == User:
+        if accessor._meta.model_name == 'user':
             ancestor_roles = accessor.roles.all()
         elif type(accessor) == Role:
             ancestor_roles = [accessor]
diff --git a/awx/main/models/rbac.py b/awx/main/models/rbac.py
index 5febf88eb2..a210a56cb1 100644
--- a/awx/main/models/rbac.py
+++ b/awx/main/models/rbac.py
@@ -170,7 +170,7 @@ class Role(models.Model):
         return reverse('api:role_detail', kwargs={'pk': self.pk}, request=request)
 
     def __contains__(self, accessor):
-        if type(accessor) == User:
+        if accessor._meta.model_name == 'user':
             return self.ancestors.filter(members=accessor).exists()
         elif accessor.__class__.__name__ == 'Team':
             return self.ancestors.filter(pk=accessor.member_role.id).exists()
```

This seems to address that particular failure case. I would like to see this patch applied and ran through the rest of the tests to see if there are any other problems hanging around.

File: awx/main/models/rbac.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def get_absolute_url(self, request=None):
         return reverse('api:role_detail', kwargs={'pk': self.pk}, request=request)
 
     def __contains__(self, accessor):
-        if type(accessor) == User:
+        if accessor._meta.model_name == 'user':
```
Comment (AlanCoding) [inline awx/main/models/rbac.py:173]: There's another case of this type checking further down in this file. In both files, actually, I think we should go ahead and remove all instances, such that the import of `User` itself is removed. That's an easy way to catch more bugs that might be caused by this.

**Review by AlanCoding — APPROVED** [review]

**Pull request merged**